drivers/led/is31fl3216a.c

```diff
@@ -0,0 +1,248 @@
+	uint8_t pwm_value;
+
+	if (led > IS31FL3216A_MAX_LEDS - 1) {
+		return -ENODEV;
+	}
+
+	if (value > 100) {
+		return -EINVAL;
+	}
+
+	pwm_value = is31fl3216a_brightness_to_pwm(value);
+	status = is31fl3216a_write_reg(&config->i2c, pwm_reg, pwm_value);
+	if (status) {
+		return status;
+	}
+
+	return is31fl3216a_update_pwm(&config->i2c);
+}
+
+static int is31fl3216a_led_on(const struct device *dev, uint32_t led)
+{
+	return is31fl3216a_led_set_brightness(dev, led, 100);
+}
+
+static int is31fl3216a_led_off(const struct device *dev, uint32_t led)
+{
+	return is31fl3216a_led_set_brightness(dev, led, 0);
+}
+
+static int is31fl3216a_init_registers(const struct i2c_dt_spec *i2c)
+{
+	int i;
+	int status;
+
+	status = is31fl3216a_write_reg(i2c, IS31FL3216A_REG_CTL_1, 0xFF);
+	if (status) {
+		return status;
+	}
+
+	status = is31fl3216a_write_reg(i2c, IS31FL3216A_REG_CTL_2, 0xFF);
+	if (status) {
+		return status;
+	}
+
+	status = is31fl3216a_write_reg(i2c, IS31FL3216A_REG_LIGHT_EFFECT, 0x00);
+	if (status) {
+		return status;
+	}
+
+	status = is31fl3216a_write_reg(i2c, IS31FL3216A_REG_CHANNEL_CONFIG,
+				       0x00);
+	if (status) {
+		return status;
+	}
+
+	status = is31fl3216a_write_reg(i2c, IS31FL3216A_REG_GPIO_CONFIG, 0x00);
+	if (status) {
+		return status;
+	}
+
+	status = is31fl3216a_write_reg(i2c, IS31FL3216A_REG_OUTPUT_PORT, 0x00);
+	if (status) {
+		return status;
+	}
+
+	status = is31fl3216a_write_reg(i2c, IS31FL3216A_REG_INT_CONTROL, 0x00);
+	if (status) {
+		return status;
+	}
+
+	status = is31fl3216a_write_reg(i2c, IS31FL3216A_REG_ADC_SAMPLE_RATE,
+				       0x00);
+	if (status) {
+		return status;
+	}
+
+	status = is31fl3216a_write_reg(i2c, IS31FL3216A_REG_FRAME_DELAY,
+				       0x00);
+	if (status) {
+		return status;
+	}
+
+	status = is31fl3216a_write_reg(i2c, IS31FL3216A_REG_FRAME_START,
+				       0x00);
+	if (status) {
+		return status;
+	}
+
+	for (i = IS31FL3216A_REG_PWM_FIRST;
+	     i <= IS31FL3216A_REG_PWM_LAST;
+	     i++) {
+		status = is31fl3216a_write_reg(i2c, i, 0);
+		if (status) {
+			return status;
+		}
+	}
+
+	status = is31fl3216a_write_reg(i2c, IS31FL3216A_REG_UPDATE, 0);
+	if (status) {
+		return status;
+	}
+
+	return is31fl3216a_write_reg(i2c, IS31FL3216A_REG_CONFIG, 0x00);
+}
+
+static int is31fl3216a_init(const struct device *dev)
+{
+	const struct is31fl3216a_cfg *config = dev->config;
+
+	LOG_DBG("Initializing @0x%x...", config->i2c.addr);
+
+	if (!device_is_ready(config->i2c.bus)) {
+		LOG_ERR("I2C device not ready");
+		return -ENODEV;
+	}
+
+	return is31fl3216a_init_registers(&config->i2c);
+}
+
+static const struct led_driver_api is31fl3216a_led_api = {
+	.set_brightness = is31fl3216a_led_set_brightness,
+	.on = is31fl3216a_led_on,
+	.off = is31fl3216a_led_off,
+	.write_channels = is31fl3216a_led_write_channels
+};
+
+#define IS31FL3216A_INIT(id) \
+	static const struct is31fl3216a_cfg is31fl3216a_##id##_cfg = {	\
+		.i2c = I2C_DT_SPEC_INST_GET(id),			\
+	};								\
+	DEVICE_DT_INST_DEFINE(						\
```

Comment:
Does it fit?

<pre>
DEVICE_DT_INST_DEFINE(id, &is31fl3216a_init, NULL, NULL,           \
		      &is31fl3216a_##id##_cfg, POST_KERNEL,                      \
		      CONFIG_LED_INIT_PRIORITY, &is31fl3216a_led_api);
</pre>

Reply:
Yes